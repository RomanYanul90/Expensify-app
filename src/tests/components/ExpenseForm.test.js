import React from 'react'
import { shallow } from 'enzyme'
import Expense_form from "../../components/Expense_form";
import expenses from "../fixtures/expenses";

test('should render Expense form correctly',()=>{
    const wrapper = shallow(<Expense_form/>)
    expect(wrapper).toMatchSnapshot()
})

test('should render Expense form with data',()=>{
    const wrapper = shallow(<Expense_form expense = {expenses[0]}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should render error msg for invalid submission',()=>{
    const wrapper = shallow(<Expense_form />)
    expect(wrapper).toMatchSnapshot()
    wrapper.find('form').simulate('submit',{
        preventDefault:()=>{}
    })
    expect(wrapper.state('error').length).toBeGreaterThan(0)
    expect(wrapper).toMatchSnapshot()
})

test('should set description on input change',()=>{
    const value = "New description"
    const wrapper = shallow(<Expense_form />)
    wrapper.find('input').at(0).simulate('change',{
        target:{ value }
    })
    expect(wrapper.state('description')).toBe(value)
})

test('should set note on textarea change',()=>{
    const value = "New note"
    const wrapper = shallow(<Expense_form />)
    wrapper.find('textarea').simulate('change',{
        target:{ value }
    })
    expect(wrapper.state('note')).toBe(value)
})

test('should set amount for valid input',()=>{
    const value = '12.22'
    const wrapper = shallow(<Expense_form />)
    wrapper.find('input').at(1).simulate('change',{
        target:{ value }
    })
    expect(wrapper.state('amount')).toBe(value)
})

test('should not set amount for valid input',()=>{
    const value = '12.223'
    const wrapper = shallow(<Expense_form />)
    wrapper.find('input').at(1).simulate('change',{
        target:{ value }
    })
    expect(wrapper.state('amount')).toBe('')
})

test('should call onSubmit prop for valid form submission ',()=>{
    const onSubmitSpy = jest.fn()
    const wrapper = shallow(<Expense_form expense = {expenses[0]} onSubmit = {onSubmitSpy}/>)
    wrapper.find('form').simulate('submit',{
        preventDefault:()=>{}
    })
    expect(wrapper.state('error')).toBe('')
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        description: expenses[0].description,
        amount: expenses[0].amount,
        note: expenses[0].note,
        createdAt: expenses[0].createdAt,
    })
})

