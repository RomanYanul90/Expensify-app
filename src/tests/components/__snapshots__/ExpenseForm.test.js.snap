// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense form correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Amount"
    type="text"
    value=""
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Add a note for your expense (optional)"
    value=""
  />
  <div>
    <button
      className="button"
    >
      Save expense
    </button>
  </div>
</form>
`;

exports[`should render Expense form with data 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value="Gum"
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Amount"
    type="text"
    value="1.95"
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Add a note for your expense (optional)"
    value=""
  />
  <div>
    <button
      className="button"
    >
      Save expense
    </button>
  </div>
</form>
`;

exports[`should render error msg for invalid submission 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Amount"
    type="text"
    value=""
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Add a note for your expense (optional)"
    value=""
  />
  <div>
    <button
      className="button"
    >
      Save expense
    </button>
  </div>
</form>
`;

exports[`should render error msg for invalid submission 2`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <p
    className="form__error"
  >
    Please provide description and amount.
  </p>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Amount"
    type="text"
    value=""
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Add a note for your expense (optional)"
    value=""
  />
  <div>
    <button
      className="button"
    >
      Save expense
    </button>
  </div>
</form>
`;
